Snake.fs

```diff
@@ -102,133 +101,115 @@ module SnakeGame =
                 |> Array.ofList
-            let food = randomFood ()
+            let food :FoodType = random.Next()
             let randomPos = freeSpaces.[random.Next(freeSpaces.Length - 1)]
             randomPos, food
 
-        let dropFood (grid:Grid) (food:Food) =
-            let (x, y), animal = food
-            grid.[x].[y] <- animal
+        let dropFood (food:Food)  (grid:Grid) =
+            let (x, y), foodType = food
+            // TODO: mutation
+            grid.[x].[y] <- CellFood(foodType)
+            grid
 
-        let slither (snake:Snake<Position>) (grid:Grid) : unit =
+        let slither (snake:Snake<Position>) (grid:Grid) : Grid =
             try
-                let rec slithering (body:Snake<Position>) =
+                let rec slithering (body:Snake<Position>) : Grid =
                     match body with
-                    | Head(p, s) -> 
+                    | Head(p, s) ->
                         let row, column = p
-                        grid.[row].[column] <- head 
+                        grid.[row].[column] <- SnakeHead
                         slithering s
-                    | Belly(p, s) -> 
+                    | Belly(p, s) ->
                         let row, column = p
-                        grid.[row].[column] <- belly 
+                        grid.[row].[column] <- SnakeBelly
                         slithering s
-                    | Tail -> ()
-                do slithering snake
+                    | Tail -> grid
+                slithering snake
             with _ -> failwith "ERROR: Could not slither snake!"
-        
-        let endGame (grid:Grid) : unit =
-            Console.Clear()
-            Array.iteri (fun i row -> 
-                Array.iteri (fun j _ ->
-                    grid.[i].[j] <- errorDisplay
-                ) row
+
+    module Graphics =
+        let private head = "🤢"
+        let private belly = "🟢"
+        let private display = "⬜"
+        let private errorDisplay = "🟥"
+
+        let private food = [|"🐁";"🐀";"🐥";"🪺";"🐸";"🐛";"🪰";"🐞";"🦗"|]
+
+        type ScreenTile = string
+        type ScreenBuffer = array<array<ScreenTile>>
+
+        // transform a logical Cell into a graphical screen tile
+        let spriteFor (cell:Cell) : ScreenTile =
+            match cell with
+                | SnakeHead -> head
+                | SnakeBelly -> belly
+                | CellFood(idx) -> food.[idx % food.Length]
+                | CellEmpty -> display
+
+        // transform a logical grid into a graphical screen buffer for display
+        let renderGrid (grid:Grid) : ScreenBuffer =
+            Array.map (fun (row:Cell array) ->
+                Array.map spriteFor row
             ) grid
-            Array.iter (fun (row:string array) -> 
-                let prettyprint = String.concat "" row
-                printfn $"{prettyprint}") grid
-            Console.Beep()
 
-    
-    module GamePlay =
+        // display a
```

Comment:
lol

Dear Past Me,
   What helpful commenting you have done.

Sincerely,
Present Me